CRM/Upgrade/Incremental/php/FiveForty.php

```diff
@@ -51,9 +51,38 @@ public function setPostUpgradeMessage(&$postUpgradeMessage, $rev) {
 
   public function upgrade_5_40_alpha1($rev) {
     $this->addTask(ts('Upgrade DB to %1: SQL', [1 => $rev]), 'runSql', $rev);
+    $this->addTask('Add option list for group_used_for', 'addGroupOptionList');
     $this->addTask('core-issue#2486  - Add product_id foreign key to civicrm_contribution_product', 'addContributionProductFK');
   }
 
+  /**
+   * @param CRM_Queue_TaskContext $ctx
+   * @return bool
+   */
+  public static function addGroupOptionList(CRM_Queue_TaskContext $ctx) {
+    $optionGroupId = \CRM_Core_BAO_OptionGroup::ensureOptionGroupExists([
+      'name' => 'note_used_for',
+      'title' => ts('Note Used For'),
+      'is_reserved' => 1,
+      'is_active' => 1,
+      'is_locked' => 1,
+    ]);
+    $values = [
+      'civicrm_relationship' => 'Relationship',
+      'civicrm_contact' => 'Contact',
+      'civicrm_participant' => 'Participant',
+      'civicrm_contribution' => 'Contribution',
+    ];
+    foreach ($values as $value => $name) {
+      \CRM_Core_BAO_OptionValue::ensureOptionValueExists([
+        'value' => $value,
+        'name' => $name,
```

Comment:
Similarly you'd want to explicitly specify label here with a ts, but not ts($name) since that's not good - maybe could expand the array above to also have entries for ts('Contact'), ts('Relationship'),.. ?